java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -264,4 +269,18 @@ private Charset getCharset(File file) {
   public boolean analysisCancelled() {
     return context.isCancelled();
   }
+
+  public void addAnalysisError(AnalysisError analysisError) {
+    analysisErrors.add(analysisError);
+  }
+
+  public void saveAnalysisErrors() {
+    if (!analysisErrors.isEmpty() && "https://sonarcloud.io".equals(context.config().get("sonar.host.url").orElse(""))) {
+      Metric.Builder metricBuilder = new Metric.Builder("sonarjava_feedback", "SonarJava feedback", Metric.ValueType.DATA);
```

Comment:
I would prefer to have the key of the metric extracted to a constant somewhere, maybe in JavaPlugin class itself.

Reply:
This was changed to use Server API thanks to @simonbrandhof feedback.